src/main.c

```diff
@@ -1,4 +1,26 @@
-int main () {
-	while(1);
+#include <stdio.h>
+#include "app/app_header.h"
+
+bool check_app_version(const app_header_t header) {
+	if (header.magic != APP_HEADER_MAGIC) {
+		return false;
+	}
+	if (header.version.major != APP_HEADER_VERSION_MAJOR) {
+		return false;
+	}
+	if (header.version.minor != APP_HEADER_VERSION_MINOR) {
+		return false;
+	}
+	return true;
+}
+
+int main() {
+	if(!(check_app_version(app_header))) {
+		while(1);
+	}
+
+	app_header.app_entry();
```

Comment:
This should go into the `boot_app()` function, and the `static` function that does this last step should be labeled a `_Noreturn` function so that the compiler can make optimizations for the steps leading up to passing control over to the app.